Sources/Ink/Internal/List.swift

```diff
@@ -16,7 +19,22 @@ internal struct List: Fragment {
 
     private static func read(using reader: inout Reader,
                              indentationLength: Int) throws -> List {
-        var list = List(isOrdered: reader.currentCharacter.isNumber)
+        let isOrdered = reader.currentCharacter.isNumber
+
+        var list: List
+        if isOrdered {
+            let startIndex = reader.currentIndex
+            defer { reader.moveToIndex(startIndex) }
```

Comment:
You shouldn't need to do this, since the `readOrRewind` function takes care of rewinding the index back to its previous state in case an error was thrown from this function.

Reply:
Unfortunately I still need to do `.moveToIndex` it since I need to revert back to the beginning of the list–but I can remove the `defer` call now, since I don't have to worry about the `try` statements.